### DISCUSSION TITLE
Colab demos for DI-engine

### DISCUSSION THREAD

**Discussion by Formulator** [opendilab/DI-engine#721] [General]
Is it possible to use DI-engine in Google Colab?  I started to explore the [hello world tutorial](https://di-engine-docs.readthedocs.io/en/latest/01_quickstart/hello_world_for_DI.html#let-the-agent-run) but found that I get dependency conflicts when calling `!pip install DI-engine` in Colab:

`ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
ipython 7.34.0 requires jedi>=0.16, which is not installed.
fiona 1.9.4.post1 requires click~=8.0, but you have click 7.1.2 which is incompatible.
dask 2023.8.1 requires click>=8.0, but you have click 7.1.2 which is incompatible.
distributed 2023.8.1 requires click>=8.0, but you have click 7.1.2 which is incompatible.
nbconvert 6.5.4 requires jinja2>=3.0, but you have jinja2 2.11.3 which is incompatible.
pip-tools 6.13.0 requires click>=8, but you have click 7.1.2 which is incompatible.
Successfully installed DI-toolkit-0.2.0 DI-treetensor-0.4.1 GitPython-3.1.35 Jinja2-2.11.3 MarkupSafe-2.0.1 URLObject-2.4.3 Werkzeug-1.0.1 bitmath-1.3.3.1 chardet-4.0.0 click-7.1.2 deprecation-2.1.0 di-engine-0.4.9 dill-0.3.7 docker-pycreds-0.4.0 easydict-1.9 enum-tools-0.10.0 farama-notifications-0.0.4 flask-1.1.4 gitdb-4.0.10 gym-0.25.1 gymnasium-0.29.1 hbutils-0.9.1 hickle-5.0.2 itsdangerous-1.1.0 mpire-2.8.0 pathtools-0.1.2 pynng-0.7.2 pytimeparse-1.1.8 redis-5.0.0 responses-0.12.1 sentry-sdk-1.30.0 setproctitle-1.3.2 setuptools-66.1.1 smmap-5.0.0 tensorboardX-2.6.2.2 treevalue-1.4.12 trueskill-0.4.5 wandb-0.15.10 yapf-0.29.0 yattag-1.15.1
WARNING: The following packages were previously imported in this runtime:
  [_distutils_hack,pkg_resources,setuptools]
You must restart the runtime in order to use newly installed versions`

I would like to suggest that the project develops some functional demos/walkthroughs/tutorials from the [dizoo](https://github.com/opendilab/DI-engine/tree/main/dizoo) in Colab notebooks.  This would get users quickly up, running and experimenting by abstracting away Python environment complexities etc.  Might be good for wider adoption?

**Comment by PaParaZz1** [comment]
Thanks for your advice, we will polish this basic demo and move more demos in dizoo to Colab.

**Comment by zjowowen** [comment]
Hi, thanks for raising issues for the quick start tutorial.

DI-engine is being frequently polished. So, there maybe some imcompatibility API in the old tutorial. Sorry for that!

For the latest quick start, you can turn to this pull request:
https://github.com/opendilab/DI-engine/pull/733/files
 
Besides, you can run dizoo demo in this way:
https://colab.research.google.com/drive/1W7a8Qf-tZT_0loeXPSngZ8uYMb593zcj

**Comment by PaParaZz1** [comment]
We have add some demos in #733, and we will add more examples in the following two months.

**Reply by Formulator** [reply]
That's great thanks, I used [3 Minutes Kickoff (colab)](https://colab.research.google.com/drive/1_7L-QFDfeCvMvLJzRyBRUW5_Q6ESXcZ4) and managed to run the code.  I would now like to now run a [gym_anytrading](https://github.com/opendilab/DI-engine/tree/main/dizoo/gym_anytrading) example but can't work out how to modify the demo for other environments!  Please could you make this one of the coming examples?  Many thanks.

**Reply by Cloud-Pku** [reply]
Building examples for gym_anytrading will be added in about one month.

**Reply by Cloud-Pku** [reply]
Hello, this is a [demo (colab)](https://colab.research.google.com/drive/1IKfjfyb0PSyTranJemvve2V38_JUs6yB?usp=sharing) for training DQN on gym-anytrading. If you have any comments, welcome to discuss.
